clients/kafka-common/src/main/java/org/eclipse/hono/kafka/client/consumer/AbstractAtLeastOnceKafkaConsumer.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.kafka.client.consumer;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.eclipse.hono.util.Lifecycle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.kafka.client.consumer.KafkaConsumer;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecords;
+
+/**
+ * A client to consume data from a Kafka cluster.
+ * <p>
+ * This consumer continuously polls for batches of messages from Kafka. Each message is passed to a message handler for
+ * processing. When all messages of a batch are processed, this consumer commits the offset and polls for a new batch.
+ * <p>
+ * This consumer provides AT LEAST ONCE message processing. When the message handler completes without throwing an
+ * exception, the message is considered to be processed and will be "acknowledged" (with the following commit). If the
+ * processing of messages or the commit of the offsets fails, other consumer instances in the same consumer group will
+ * consume the messages from the last committed offset. Already processed messages will then be again in the next batch
```

Comment:
> other consumer instances in the same consumer group will consume the messages from the last committed offset

This only happens because in case of the commit error, the consumer is stopped here, right? Without that information, the sentence is a bit confusing here.

Reply:
Please have a look if it is now more understandable.